css-shared-element-transitions/index.bs

```diff
@@ -324,103 +320,124 @@ A {{Document}} additionally has:
+## Additions to {{Document}} ## {#additions-to-document-api}
 
 <script type=idl>
-interface SameDocumentTransition {
-    constructor();
-    Promise<undefined> prepare(PrepareCallback cb);
-    undefined abandon();
-    readonly attribute Promise<undefined> finished;
+partial interface Document {
+    DOMTransition createTransition(DOMTransitionInit init);
 };
 
-callback PrepareCallback = Promise<any> ();
+dictionary DOMTransitionInit {
+    ChangeDOMCallback changeDOM;
+};
+
+callback ChangeDOMCallback = Promise<any> ();
+</script>
+
+### {{Document/createTransition()}} ### {#DOMTransition-prepare}
+
+<div algorithm>
+    The [=method steps=] for
+    <dfn method for=Document>createTransition(|init|)</dfn>
+    are as follows:
+
+    1. Let |transition| be a new {{DOMTransition}} object in [=this's=] [=relevant Realm=].
+
+    1. Set |transition|'s [=DOMTransition/DOM change callback=] to |init|[{{DOMTransitionInit/changeDOM}}].
+
+    1. Let |document| be [=this's=] [=relevant global object's=] [=associated document=].
+
+    1. Let |transitionsToSkip| be a [=/list=] of every {{DOMTransition}}
+        within [=this's=] [=relevant Realm=],
+        that has not [=DOMTransition/concluded=],
+        and is not |transition|.
+
+    1. Assert: The [=list/size=] of |transitionsToSkip| is 0 or 1.
+
+    1. [=list/For each=] |transitionToSkip| of |transitionsToSkip|, [=skip the page transition=] |transitionToSkip| with an "{{AbortError}}" {{DOMException}} in [=this's=] [=relevant Realm=].
+
+    1. [=Assert=]: |document|'s [=document/pending same document transition outgoing capture=] is null.
+
+    1. Set |document|'s [=document/pending same document transition outgoing capture=] to |transition|.
+
+        Note: This process continues in [=perform an outgoing capture=].
+
+    1. Return |transition|.
+
+</div>
+
+## The {{DOMTransition}} interface ## {#the-domtransition-interface}
+
+<script type=idl>
+interface DOMTransition {
+    undefined skipTransition();
+    readonly attribute Promise<undefined> finished;
+    readonly attribute Promise<undefined> ready;
+    readonly attribute Promise<undefined> domUpdated;
+};
 </script>
 
 <div class="note">
-    The {{SameDocumentTransition}} represents and controls
+    The {{DOMTransition}} represents and controls
     a single same-document transition. That is, it controls a transition where the
     starting and ending document are the same, possibly with changes to the
     document's DOM structure.
 </div>
 
-{{SameDocumentTransition}} objects have the following:
+A {{DOMTransition}} has the following:
 
-<dl dfn-for="SameDocumentTransition">
+<dl dfn-for="DOMTransition">
     : <dfn>tagged elements</dfn>
     :: a [=/map=],
         whose keys are [=page transition tags=] and whose values are [=captured elements=].
         Initially a new [=map=].
 
-    : <dfn>phase</dfn>
-    :: "`idle`", "`preparing`", "`running`", or "`finished`".
-        Initially "`idle`".
+    : <dfn>concluded</dfn>
```

Comment:
I think this is fine, but why not something like "done" or "finished"? (https://www.w3.org/TR/design-principles/#naming-common-words)

Reply:
I thought "done" and "finished" imply 'successfully' whereas "concluded" didn't. I could use "settled" which is the language promises use.

Reply:
Makes sense, I don't really have much of a preference just wanted to see if there was a rationale for using 'concluded'